### PR TITLE
Add swiftlint linter

### PR DESCRIPTION
**Pull request by brendanjerwin**
https://github.com/realm/SwiftLint

### CODE REVIEW

File: autoload/neomake/makers/ft/swift.vim [hunk 1]
```diff
@@ -0,0 +1,11 @@
+function! neomake#makers#ft#swift#EnabledMakers()
+    return ['swiftlint']
+endfunction
+
+function! neomake#makers#ft#swift#swiftlint()
+    return {
+        \ 'args': ['lint', '--config', './.swiftlint.yml', '--quiet'],
```
Comment (brendanjerwin) [inline autoload/neomake/makers/ft/swift.vim]: Is this a reasonable way to deal with the config file? Is there something better?

File: autoload/neomake/makers/ft/swift.vim [hunk 2]
```diff
@@ -5,7 +5,7 @@ endfunction
 function! neomake#makers#ft#swift#swiftlint()
     return {
         \ 'args': ['lint', '--config', './.swiftlint.yml', '--quiet'],
-        \ 'errorformat': '%f:%l:%c: %trror: %m,%f:%l:%c: %tarning: %m',
+        \ 'errorformat': '%f:%l:%c: %trror: %m,%f:%l:%c: %tarning: %m,%f:%l: %tarning: %m,%f:%l: %trror: %m',
```
Comment (brendanjerwin) [inline autoload/neomake/makers/ft/swift.vim]: Yes. Yes it is.

File: autoload/neomake/makers/ft/swift.vim (hunk 1 above)
Comment (blueyed) [inline autoload/neomake/makers/ft/swift.vim]: Depends on the tool.
Wouldn't that be used by default? (i.e. leave it out here)

Comment (brendanjerwin) [inline autoload/neomake/makers/ft/swift.vim]: Yeah... I thought I had trouble with it if I didn't specify, but the linter _should_ find its own config. I'll give it another go and see.

**Comment by blueyed** [comment]
See #412 also.

**Comment by blueyed** [comment]
@brendanjerwin 
ping.

#412 is merged now.

**Comment by blueyed** [comment]
@brendanjerwin 
Ping.
Are you still with us?

**Comment by brendanjerwin** [comment]
@blueyed sorry, no. I've moved on to another gig and don't use Swift anymore.

**Comment by blueyed** [comment]
Closing it then.
It can be picked up by somebody else, of course.

**Pull request closed without merging**